mdx-web/src/test/groovy/com/mx/path/model/mdx/web/controller/AuthenticationControllerTest.groovy

```diff
@@ -497,6 +498,54 @@ class AuthenticationControllerTest extends Specification implements WithMockery
     response.statusCode == HttpStatus.NO_CONTENT
   }
 
+  def "unlockUser"() {
+    given:
+    AuthenticationController.setGateway(gateway)
+    def unlockUser = new UnlockUser()
+    doReturn(new AccessorResponse<UnlockUser>().withResult(unlockUser).withStatus(PathResponseStatus.OK)).when(id).unlockUser(unlockUser)
+
+    when:
+    def response = subject.unlockUser(unlockUser)
+
+    then:
+    verify(gateway).id()              || true
+    verify(id).unlockUser(unlockUser) || true
+    Session.current().getId() == response.getHeaders().getFirst("mx-session-key")
+    response.statusCode == HttpStatus.ACCEPTED
+  }
+
+  def "unlockUser - ACCEPTED"() {
+    given:
+    AuthenticationController.setGateway(gateway)
+    def challenge = getChallenge()
+    def unlockUser = new UnlockUser().tap { challenges = [challenge] }
+    def expected = new UnlockUser().tap { challenges = [new Challenge()] }
```

Comment:
These tests function as a spec, I wouldn't return an empty Challenge in this instance because that is not expected by the spec. Consider populating the returned challenge (could do this easily w/ data already in the spec).

This is done in some tests at this layer but not everywhere. These values don't necessarily get tested against (these tests mainly verify behavior), but there is actually no instance when unlockUser would return a 202 and an empty Challenge...

Reply:
@meotch, this is ready for review.